R/shiny.R

```diff
@@ -0,0 +1,25 @@
+.globals <- new.env(parent = emptyenv())
+
+shiny_options_set <- function() {
+  do_shiny({
+    .globals$shinyOptions <- shiny::shinyOptions()
+    shiny::shinyOptions(bootstraplib = TRUE)
+  })
+}
+
+shiny_options_restore <- function() {
+  do_shiny({
+    shiny::shinyOptions(bootstraplib = .globals$shinyOptions$bootstraplib)
+  })
+}
+
+do_shiny <- function(expr) {
+  if ("shiny" %in% loadedNamespaces()) {
+    return(expr)
+  }
+
+  setHook(
```

Comment:
I think that calling `setHook` this way has a couple issues: it could create an indefinitely large queue of callbacks to call when shiny is loaded. If the user has no intention of even using Shiny in this session, that is an unnecessary use of memory.

The other issue is that if/when shiny actually is loaded, this will execute many of these stored-up callbacks, which isn't necessary. We really only care about making the most recent bootstrap theme used by shiny.

Probably better do something like this: in `.onLoad`, check if shiny is loaded; if not, call `setHook()`. What that hook will do (if shiny is loaded later) is call call `shinyOptions()`.